src/slideshow.c

```diff
@@ -244,9 +246,26 @@ void feh_reload_image(winwidget w, int resize, int force_new)
 	return;
 }
 
+static void *thread_start(void *arg)
+{
+	feh_file *file = (feh_file*)arg;
+
+	// Hack to prevent segfault when scrolling quickly through already cached images
+	nanosleep((const struct timespec[]){{0, 1000000L}}, NULL);
```

Comment:
Why is that necessary, there should probably a better fix available?

Reply:
I don't know to be honest. I guess imlib2 is not thread safe.

Reply:
Yeah it absolutely is not thread safe. Even if there's a lock around this one instance, all other imlib operations would also need it. Or maybe put all of them into a separate worker thread. In any case, that's all a bit more work.

Reply:
I don't think that we can work around this. Proper preloading in the background will always require concurrent imlib calls.